src/monitor/circuit_graph.rs

```diff
@@ -0,0 +1,379 @@
+use std::{
```

Comment:
I would document this file

Reply:
People may be confused by the fact that you have two graphs: one is the DBSP circuit, and the other one is this topological representation. In Java I had the circuit itself have methods to dump the topology.
I am still not sure why you chose to separate them.